commonutils/src/main/java/io/intrepid/commonutils/StringUtils.java

```diff
@@ -0,0 +1,504 @@
+     * Items that are empty or null are skipped entirely (i.e. there won't be two or more separators directly
+     * following each other).
+     * <p/>
+     * E.g.
+     * separateItemsWith(", ", "Hello", "This", null, "is", "", "my name")
+     * will return
+     * "Hello, This, is, my name"
+     *
+     * @param separator The string with which to separate the items.
+     * @param items     The items that appear in the list.
+     * @return The String of items.
+     */
+    public static CharSequence separateItemsWith(CharSequence separator, CharSequence... items) {
+        StringBuilder sb = new StringBuilder();
+        for (CharSequence item : items) {
+            if (TextUtils.isEmpty(item)) {
+                continue;
+            }
+
+            if (sb.length() == 0) {
+                sb.append(item);
+            } else {
+                sb.append(separator).append(item);
+            }
+        }
+        return sb;
+    }
+
+    /**
+     * @param separator The string with which to separate the items.
+     * @param items     The items that appear in the list.
+     * @return The String of items.
+     * @see #separateItemsWith(CharSequence, CharSequence...)
+     */
+    public static CharSequence separateItemsAsStringsWith(CharSequence separator, List<?> items) {
+        if ((items == null) || items.isEmpty()) {
+            return "";
+        }
+
+        CharSequence[] itemsArray = new CharSequence[items.size()];
+        for (int i = 0; i < items.size(); i++) {
+            Object item = items.get(i);
+            itemsArray[i] = (item != null) ? items.get(i).toString() : null;
+        }
+        return separateItemsWith(separator, itemsArray);
+    }
+
+    /**
+     * @param separator The string with which to separate the items.
+     * @param items     The items that appear in the list.
+     * @return The String of items.
+     * @see #separateItemsWith(CharSequence, CharSequence...)
+     */
+    public static CharSequence separateItemsWith(CharSequence separator, List<? extends CharSequence> items) {
+        if ((items == null) || items.isEmpty()) {
+            return "";
+        }
+
+        CharSequence[] itemsArray = items.toArray(new CharSequence[items.size()]);
+        return separateItemsWith(separator, itemsArray);
+    }
+
+    /**
+     * Repeats the input string a number of times, each repetition separated from the other by the separator.
+     * The returned string will never start or end with the given separator unless the input string is empty.
+     *
+     * @param string      The string to repeat.
+     * @param separator   The separator between repetitions.
+     * @param repeatCount The times that the given string should be repeated.
+     * @return The repeated string.
+     */
+    public static CharSequence repeat(CharSequence string, CharSequence separator, int repeatCount) {
+        if (repeatCount < 1) {
+            return null;
```

Comment:
Not sure if I have a strong opinion, but would it be better to pass an empty string instead of null here?